jax/_src/lib/xla_bridge.py

```diff
@@ -340,7 +329,10 @@ def backends():
           logging.info("Unable to initialize backend '%s': %s", platform,
                        err)
           _backends_errors[platform] = str(err)
```

Comment:
We only need to populate _backend_errors in the else branch

Reply:
Oh sorry, I mean in the else branch below. Basically we don't want to change the existing behavior when --jax_platforms isn't set, so you still want to do the logging + populating _backends_errors as an alternative to raising a RuntimeError. You can also pull out creating the "Unable to initialize backend" string since you'll use that in both places below.

Reply:
But the logging does not happen! if we add or not add that line with LOG(INFO) it is not printing the message. That is why I removed it, without populating the backends the tests failed, so I brought it back :)

Reply:
That's because jax doesn't print INFO logs by default, but you can enable them with TF_CPP_MIN_LOG_LEVEL=0. This log message actually ends up confusing people because importing tensorflow also turns on logging, so people see the "tpu_driver" initialization failing (which is the legacy TPU backend, not the TPU VM backend) and reasonably think it will affect the TPU VM. So there is an argument for removing it now that we can get the error with JAX_PLATFORMS (being able to access the error is super important for debugging). I think we should keep it for now though, because it might still be more convenient than figuring out which platform to specify in some cases.

Please move these into the else clause below too, none of this needs to happen if we're about to raise a RuntimeError.

Reply:
The _backends_errors bit still needs to go in the else clause (since it's not necessary otherwise). Otherwise LGTM!